src/clr-angular/data/stack-view/stack-block.service.ts

```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class StackBlockService {
+  public id;
```

Comment:
This is related to what I mentioned earlier. Users won't be able to set their own ids, as the element we set the unique id on is wrapped inside our component template. So If users don't need to set their own ids, we don't need this service the id.

As the stackblock component injects the unique id provider, all its children can inject and access the autogenerated id.

Reply:
Good catch, I had bigger plans originally but this service doesn't need to exist.